documentation/modules/ROOT/pages/configuration/storage.adoc

```diff
@@ -0,0 +1,705 @@
+|[[offset-storage-redis-key]]<<offset-storage-redis-key, `offset.storage.redis.key`>>
+|metadata:debezium:offsets
+|The Redis key that will be used to store the offsets.
+|[[offset-storage-redis-address]]<<offset-storage-redis-address, `offset.storage.redis.address`>>
+|No default
+|The url that will be used to access Redis.
+|[[offset-storage-redis-user]]<<offset-storage-redis-user, `offset.storage.redis.user`>>
+|No default
+|The user that will be used to access Redis.
+|[[offset-storage-redis-password]]<<offset-storage-redis-password, `offset.storage.redis.password`>>
+|No default
+|The password that will be used to access Redis.
+|[[offset-storage-redis-db-index]]<<offset-storage-redis-db-index, `offset.storage.redis.db.index`>>
+|0
+|The database index (0..15) that will be used to access Redis.
+|[[offset-storage-redis-ssl-enabled]]<<offset-storage-redis-ssl-enabled,`offset.storage.redis.ssl.enabled`>>
+|false
+|Use SSL for Redis connection.
+|[[offset-storage-redis-connection-timeout-ms]]<<offset-storage-redis-connection-timeout-ms, `offset.storage.redis.connection.timeout.ms`>>
+|2000
+|Connection timeout (in ms).
+|[[offset-storage-redis-socket-timeout-ms]]<<offset-storage-redis-socket-timeout-ms, `offset.storage.redis.socket.timeout.ms`>>
+|2000
+|Socket timeout (in ms).
+|[[offset-storage-redis-retry-initial-delay-ms]]<<offset-storage-redis-retry-initial-delay-ms, `offset.storage.redis.retry.initial.delay.ms`>>
+|300
+|Initial retry delay (in ms).
+|[[offset-storage-redis-retry-max-delay-ms]]<<offset-storage-redis-retry-max-delay-ms, `offset.storage.redis.retry.max.delay.ms`>>
+|10000
+|Maximum retry delay (in ms).
+|[[offset-storage-redis-retry-max-attempts]]<<offset-storage-redis-retry-max-attempts, `offset.storage.redis.retry.max.attempts`>>
+|10
+|Maximum number of retry attempts before giving up.
+|[[offset-storage-redis-wait-enabled]]<<offset-storage-redis-wait-enabled,`offset.storage.redis.wait.enabled`>>
+|false
+|Enables wait for replica. In case Redis is configured with a replica shard, this allows to verify that the data has been written to the replica.
+|[[offset-storage-redis-wait-timeout-ms]]<<offset-storage-redis-wait-timeout-ms,`offset.storage.redis.wait.timeout.ms`>>
+|1000
+|Timeout when wait for replica.
+|[[offset-storage-redis-wait-retry-enabled]]<<offset-storage-redis-wait-retry-enabled, `offset.storage.redis.wait.retry.enabled`>>
+|false
+|Enables retry on wait for replica failure.
+|[[offset-storage-redis-wait-retry-delay-ms]]<<offset-storage-redis-wait-retry-delay-ms, `offset.storage.redis.wait.retry.delay.ms`>>
+|1000
+|Delay of retry on wait for replica failure.
+
+|===
+
+=== Internal schema history store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[schema-history-internal-redis]]<<schema-history-internal-redis, `schema.history.internal`>>
+|No default
+|Must be set to `io.debezium.storage.redis.history.RedisSchemaHistory`
```

Comment:
If this and similar properties have required values, can they be coded to carry the appropriate defaults?